docs/content/v0.20/architecture.md

```diff
@@ -1,36 +1,479 @@
+          placement name (note that placement is cluster-scoped) and the WDS name. 
+          This way, when a placement is deleted or updated it is possible to 
+          locate the associated ManifestWorks for deletion or label removal 
+          (if more than one label is present, as other placements may “own” the object).
+        - For each cluster:
+          -  Sets the manifest namespace == name of the cluster
+          -  Uses the client connected to the ITS to do a server-side
+              apply of the manifest into the namespace.
+        - Worker returns and is ready to pick other keys from the queue.
+
+There are other event flows, based on the object GVK and type of event.
+The following sections broadly describe these flows:
+
+#### Object Deleted 
+
+When an object is deleted from the WDS, the handler’s *DeleteFunc* is
+invoked. Since most objects do not have a finalizer (which prevents
+deletion from the API server until the controller that sets the
+finalizer does some cleanup and then removes the object), a shallow copy
+of the object is added to a field in the key before pushing it to the
+work queue. Then:
+
+- Worker pulls the key from the work queue
+
+- Flow continues the same way as in the create/update scenario, however
+  the deletedObject field in the key indicates that the object has been
+  deleted and that it needs to be removed from all clusters.
+
+- Running the selectors matching logic as in the create/delete scenario
+  once again produces a list of clusters.
+
+- Worker iterates over the list of clusters and deletes the manifest
+  work for the object from each namespace associated with the cluster.
+
+- Worker returns and is ready to pick other keys from the queue.
+
+#### Placement Created or Updated 
+
+Worker pulls key from queue; if it is a placement and it has not been
+deleted (deletion timestamp not set) it follows the following flow:
+
+- Re-enqueues all objects to force re-evaluation: this is done by 
+  iterating all GVK-indexed listers, listing objects for each lister
+  and re-enqueuing the key for each object.
+- Remove ManifestWorks from ITS for objects no longer matching: generate
+  the ManagedByPlacementLabelKey for the current (Placement, WDS) and use
+  that to retrieve all the manifestworks associated with the (Placement, WDS).
+  Then, for each manifestwork, extract the wrapped object, and re-evaluate the
+  object vs. the current placement. If no longer a match (either for the
+  "what" or the "where" part) check if each manifestwork has other (Placement, WDS)
+  labels. If yes, remove the label, if not, delete the manifestwork.
+
+Re-enqueuing all object keys forces the re-evaluation of all objects vs.
+all placements. This is a shortcut as it would be more efficient to
+re-evaluate all objects vs. the changed placement only, but it saves
+some additional complexity in the code.
+
+#### Placement Deleted
+
+When a new placement is created, the placement controller sets a finalizer
```

Comment:
s/When a new placement is created/When the placement controller first processes a new Placement/

Reply:
Done